### PR TITLE
Feature/wmt 3072 reroute transto gwfield

### PR DESCRIPTION
**Pull request by ocallaghanit**
# Summary
[ ] Bug fix ( non-breaking change which fixes issue)
[ ] New Feature ( non-breaking change which adds functionality )
[ ] Breaking Change (fix or feature that would cause existing functionality to change)

- Link to issue in Jira 

# Proposed Changes
- Changes made to the code - brief summary 

# Checklist
- [x] Unit Test written 
- [ ] Acceptance Tests updated
- [ ] Documentation Updated 
- [ ] Dependencies updated?  ( services, libraries)?
- [ ] SonarLint -  ( use the plugin )

# Additional Info
- How to demonstrate functionality  

# Screenshots

- Screen shot of Passed Unit/Acceptance Test

### CODE REVIEW

File: rabbitmq/rm-rabbit-definition.yml [hunk 1]
```diff
@@ -1,11 +1,18 @@
 {
-  "queues":[
+  "queues": [
     {
       "name": "GW.Permanent.ErrorQ",
```
Comment (bobby-ons) [inline rabbitmq/rm-rabbit-definition.yml:4]: We need to remove this GW.Permanent.ErrorQ from this rm-rabbit-definition file

File: src/main/java/uk/gov/ons/census/fwmt/jobservice/config/JobsConfig.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+@Configuration
+@EnableScheduling
+public class JobsConfig {
+  @Autowired
+  private QueueMigrator queueMigrator;
+
+  private String originQ = "GW.Transient.ErrorQ";
+  private String destRoute = "";
+
+  @Scheduled(fixedDelay = 1000)
```
Comment (bobby-ons) [inline src/main/java/uk/gov/ons/census/fwmt/jobservice/config/JobsConfig.java]: Can we make this fixedDelay of the spring scheduler to be configurable?

File: src/main/java/uk/gov/ons/census/fwmt/jobservice/rabbit/TransientExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package uk.gov.ons.census.fwmt.jobservice.rabbit;
+
```
Comment (bobby-ons) [inline src/main/java/uk/gov/ons/census/fwmt/jobservice/rabbit/TransientExceptionHandler.java:2]: This class does not have my full changes as I changed MAX_RETRY_COUNT to be a configurable value and some other changes.

File: src/test/java/uk/gov/ons/census/fwmt/jobservice/rabbit/TransientExceptionHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
+package uk.gov.ons.census.fwmt.jobservice.rabbit;
```
Comment (bobby-ons) [inline src/test/java/uk/gov/ons/census/fwmt/jobservice/rabbit/TransientExceptionHandlerTest.java:1]: This test does not include my latest changes.

**Review by bobby-ons — COMMENTED** [review]
I reviewed with few comments

File: src/main/java/uk/gov/ons/census/fwmt/jobservice/rabbit/RmReceiver.java [hunk 5]
```diff
@@ -92,14 +94,17 @@ public void receiveCreateMessage(FwmtActionInstruction rmRequest, @Header("times
       default:
         break; //TODO THROW ROUTUNG FAILURE
       }
+    } catch (RestClientException e) {
+      log.error("- Create Message - Error sending message - {}  error - {} ", rmRequest, e.getMessage(), e);
+      transientExceptionHandler.handleMessage(message);
     } catch (Exception e) {
       log.error("- Create Message - Error sending message - {}  error - {} ", rmRequest, e.getMessage(), e);
-      gatewayRabbitTemplate.convertAndSend("GW.Error.Exchange", "gw.receiver.error", message);
+      gatewayRabbitTemplate.convertAndSend("GW.Error.Exchange", "gw.permanent.error", message);
```
Comment (NickGrantX) [inline src/main/java/uk/gov/ons/census/fwmt/jobservice/rabbit/RmReceiver.java:102]: I don't work on your team, but am following this with interest. I would say that RM regularly self-heals from a number of glitches (network outages, zone outages, Rabbit node problems, DB outages) and we don't ever consider anything to be a "permanent" error until a person has reviewed it... often times we fix some underlying infrastructure problem (or it's fixed by Google) and then our system self-heals.

Just my unsolicited 2 cents.

**Review by NickGrantX — COMMENTED** [review]
More unsolicited thoughts from somebody outside your team, so feel free to ignore, but queue operations such as consuming a message from one queue and publishing a message to a different queue, would usually have a transaction wrapper around the operation, to safeguard against data loss. Rabbit and Spring support transactions very well.

File: src/main/java/uk/gov/ons/census/fwmt/jobservice/rabbit/RmReceiver.java (hunk 5 above)
Comment (ocallaghanit) [reply src/main/java/uk/gov/ons/census/fwmt/jobservice/rabbit/RmReceiver.java:102]: Hey @NickGrantONS appreciate your input, for me the permanent queue is mainly about covering if there are any issues with the data, in a perfect world the data we receive should be fine, but obviously if there are any issues with it then we'd like to handle it again, maybe we find a bug in our code, or we need to apply a fix? If all goes well then this queue will never be used.

Comment (sdiaz1000) [reply src/main/java/uk/gov/ons/census/fwmt/jobservice/rabbit/RmReceiver.java:102]: @NickGrantONS feel free to being nosey

**Pull request merged**